redaxo/src/core/lib/sql/sql.php

```diff
@@ -829,6 +829,30 @@ public function insert()
         return $this;
     }
 
+    public function insertOrUpdate()
+    {
+        if ($this->records) {
+            return $this->setMultiRecordQuery('INSERT', true);
```

Comment:
Warum weiß man dass es an dieser stelle kein Update sein kann?

Reply:
Hmm d.h. Man kann jetzt auch ein UPDATE on duplicate UPDATE generieren? :-)

Reply:
Ggf. Würd ich die api ändern, ist bissl verwirrend zu lesen. Vllt ne eigene methode statt bool flag?

Reply:
Theoretisch ja. Wobei die Methode `private` ist, daher fand ich es so ok. Bin aber für bessere Vorschläge offen.

Reply:
SetQuery aus setMultiRecordQuery() entfernen und in den caller verlagern. Stattdessn nur den query bauen

Reply:
Hatte ich auch überlegt, wobei dann die Methode die Query und die Params liefern muss, also zum Beispiel als Array `[$query, $params]`. Können wir so machen.

Reply:
Passt für mich

Reply:
Ach ich hatte es auch deswegen nicht gemacht, damit ich nicht erneut alle vorkommenden Fields ermitteln muss, was bereits in `setMultiRecordQuery` geschieht.
Also dieser Part: https://github.com/redaxo/redaxo/blob/master/redaxo/src/core/lib/sql/sql.php#L1508-L1519

Das bräuchte man erneut, um die Felder für die `ON DUPLICATE KEY UPDATE`-Anweisung zu bekommen.